types/react-native/index.d.ts

```diff
@@ -5415,10 +5415,7 @@ export namespace StyleSheet {
      * their respective objects, merged as one and then returned. This also explains
      * the alternative use.
      */
-    export function flatten<T>(style?: RegisteredStyle<T>): T;
-    export function flatten(style?: StyleProp<TextStyle>): TextStyle;
-    export function flatten(style?: StyleProp<ImageStyle>): ImageStyle;
-    export function flatten(style?: StyleProp<ViewStyle>): ViewStyle;
+    export function flatten<T>(style?: StyleProp<T>): T;
```

Comment:
One example where this won't work is when the passed in argument is of type `StyleProp<TextStyle | ImageStyle>` in which case this would return `TextStyle | ImageStyle` instead of the actual `TextStyle & ImageStyle`. But I think this is better than what was there before.